docs/language-server.md

```diff
@@ -0,0 +1,137 @@
+# Language Server
+
+In order to support Rego policy development in editors like
+[VS Code](https://github.com/open-policy-agent/vscode-opa) or [Zed](https://github.com/StyraInc/zed-rego),
+Regal provides an implementation of the
+[Language Server Protocol](https://microsoft.github.io/language-server-protocol/) (LSP) for Rego.
+
+This implementation allows the result of linting to be presented directly in your editor as you work on your policies,
+and without having to call Regal from the command line. The language server however provides much more than just
+linting!
+
+## Features
+
+The Regal language server currently supports the following LSP features:
+
+### Diagnostics
+
+Diagnostics are errors, warnings, and information messages that are shown in the editor as you type. Regal currently
+uses diagnostics to present users with either parsing errors in case of syntax issues, and linter violations reported
+by the Regal linter.
+
+<img
+  src="assets/lsp/diagnostics.png"
+  alt="Screenshot of diagnostics as displayed in Zed"/>
+
+Future versions of Regal may include also compilation errors as part of diagnistics messages.
```

Comment:
Worth a link to https://github.com/StyraInc/regal/issues/745?

Reply:
For sure! Fixing.